app/js/lib/intent-parser/index.js

```diff
@@ -0,0 +1,28 @@
+import ReminderParser from './reminder-parser';
+import QueryParser from './query-parser';
+
+const p = Object.freeze({
+  // Properties
+  parsers: Symbol('parsers'),
+});
+
+export default class IntentParser {
+  constructor() {
+    this[p.parsers] = [
+      new ReminderParser(),
+      new QueryParser(),
+    ];
+
+    window.intentParser = this;
+  }
+
+  parse(phrase = '') {
+    const promises = this[p.parsers]
+      .map((parser) => parser.parse(phrase));
+
+    // Return the first non null parsed result.
+    return Promise.all(promises)
```

Comment:
Took this discussion offline and agreed this is a big change that'll needed in the future. For now, let's land it as is.